src/psyclone/psyir/nodes/member.py

```diff
@@ -0,0 +1,159 @@
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author: A. R. Porter, STFC Daresbury Lab
+# -----------------------------------------------------------------------------
+
+''' This module contains the implementation of the Member node.'''
+
+from __future__ import absolute_import
+import six
+from psyclone.psyir.nodes.node import Node
+from psyclone.psyir.symbols import TypeSymbol
+from psyclone.psyir.symbols.datatypes import StructureType
+
+
+class Member(Node):
+    '''
+    Node representing a membership expression of a structure.
+    As such it is a leaf in the PSyIR tree.
+
+    :param struct_type: the datatype of the structure containing the member \
+                        that is being referenced.
+    :type struct_type: :py:class:`psyclone.psyir.symbols.StructureType` or \
+                       :py:class:`psyclone.psyir.symbols.TypeSymbol`
+    :param str member: the member of the 'struct_type' structure that is \
+                       being referenced.
+    :param parent: the parent of this node in the PSyIR tree.
+    :type parent: :py:class:`psyclone.psyir.nodes.StructureReference` or \
+                  :py:class:`psyclone.psyir.nodes.Member`
+
+    :raises TypeError: if the supplied struct_type, member or parent are of \
+                       the wrong type.
+    :raises NotImplementedError: if the supplied struct_type doesn't have an \
+                                 associated type definition.
+    :raises TypeError: if the supplied struct_type does not contain the \
+                       specified member.
+
+    '''
+    # Textual description of the node.
+    _children_valid_format = "<LeafNode>"
+    _text_name = "Member"
+    _colour_key = "Reference"
+
+    def __init__(self, member, struct_type=None, parent=None):
```

Comment:
Yes, that was a hangover from my old way of doing it. I left it there because I thought (without really actually thinking about it) that it might be needed in future. I've removed it now. I like the idea of computing the structure type rather than storing it. I think that can be done in a future PR.